### PR TITLE
Ensure the sts or deployment created and add null check for advancedconfig 

### PR DESCRIPTION
**Pull request by hchenxa**
Signed-off-by: hchenxa <huichen@redhat.com>

### CODE REVIEW

File: tests/pkg/utils/mco_deploy.go [hunk 1]
```diff
@@ -363,6 +363,10 @@ func CheckMCOComponents(opt TestOptions) error {
 			return err
 		}
 
+		if len((*deployList).Items) == 0 {
+			return fmt.Errorf("should have deployment created with label %s", deploymentLabel)
+		}
+
 		for _, deployInfo := range (*deployList).Items {
```
Comment (clyang82) [inline tests/pkg/utils/mco_deploy.go:370]: 💯 Thanks for merging it

File: tests/pkg/utils/mco_deploy.go [hunk 2]
```diff
@@ -745,6 +753,15 @@ func GetMCOAddonSpecResources(opt TestOptions) (map[string]interface{}, error) {
...
 	}
 
+	spec := mco.Object["spec"].(map[string]interface{})
+	if _, addonSpec := spec["observabilityAddonSpec"]; !addonSpec {
+		return nil, fmt.Errorf("the MCO CR did not have observabilityAddonSpec spec configed")
+	}
+
+	if _, resSpec := mco.Object["spec"].(map[string]interface{})["observabilityAddonSpec"].(map[string]interface{})["resources"]; !resSpec {
+		return nil, fmt.Errorf("the MCO CR did not have observabilityAddonSpec.resources spec configed")
+	}
```
Comment (clyang82) [inline tests/pkg/utils/mco_deploy.go:763]: s/mco.Object["spec"].(map[string]interface{})/spec/

Comment (hchenxa) [reply tests/pkg/utils/mco_deploy.go:763]: updated

**Pull request merged**